tensorflow/python/kernel_tests/metrics_test.py

```diff
@@ -3331,7 +3331,7 @@ def testMultipleUpdatesWithMissingClass(self):
       sess.run(variables.local_variables_initializer())
       for _ in range(5):
         sess.run(update_op)
-      desired_output = np.mean([1.0 / 3.0, 2.0 / 4.0, 0.])
```

Comment:
Why does this change?

Reply:
This test is using 3 classes but there are only predictions/labels for the first 2 classes. This means the last row and col of the matrix will be zero. Previously this would cause the result to be 0 but now those results are simply ignored.

Reply:
Should those results still be zero? Otherwise it's an API compatibility change and we need to make a new v2 version of this metric and initially put it in contrib.

Reply:
As far as I understand this, the matrix stays the same (the values are still 0). All that changes is that we count how many classes are actually in the input and divide by that number instead of the matrix size. 
In this test case the output changes because it is no longer:
1/3 * (1.0 / 3.0 +  2.0 / 4.0 + 0.) but instead
1/2 * (1.0 / 3.0 +  2.0 / 4.0 + 0.)
Even though the matrix dimensions do not change.

Reply:
Let's add an iou v2 metric in contrib then.

Reply:
Are you sure? Sorry if I wasn't clear in my previous comment:
All that changes is the final result of the function in a corner case where one (or more) of the classes are not present. The matrix stays the same.

I can move it to contrib but I assumed that such a small change would not require a v2.

Reply:
Ok, I see.